### PR TITLE
fix(formly-form): Issue number 542

### PR DESCRIPTION
**Pull request by MCKRUZ**
Fixed issue with watches not being set on deep links

### CODE REVIEW

**Comment by MCKRUZ** [comment]
Did this go live yet?

File: src/directives/formly-form.test.js [hunk 1]
```diff
@@ -933,12 +933,19 @@ describe('formly-form', () => {
 
       const listener = sinon.spy()
       const expression = sinon.spy()
+      const deepLinkField = getNewField()
+      deepLinkField.key = 'foo.bar'
+      deepLinkField.watcher = {
+        listener,
+        expression: sinon.spy(),
```
Comment (kentcdodds) [inline src/directives/formly-form.test.js]: Sorry, I think I told you wrong. The `expression` shouldn't be here. We want to test behavior for when an expression is omitted. So this should be:

```javascript
deepLinkField.watcher = {
  listener: sinon.spy(),
  // no expression
}
```

Then we need to manually change the value of `foo.bar`, run `scope.$digest()` and validate that the `listener` was called with the new value of `foo.bar`. I suggest that this all be put into its own `it` statement.

Comment (MCKRUZ) [inline src/directives/formly-form.test.js]: Does this do the trick?

it(`should add watches on deep dive fields`, () => {
      debugger
      const formWithOptions = '<formly-form model="model" fields="fields" options="options"></formly-form>'
      scope.model = {}
      scope.options = {}

      const deepLinkField = getNewField()
      deepLinkField.key = 'foo.bar'
      deepLinkField.watcher = {
        listener: sinon.spy(),
      }

      scope.fields = [deepLinkField]
      compileAndDigest(formWithOptions)
      expect(compileAndDigest).to.not.throw()
      expect(deepLinkField.watcher.listener).to.have.been.called
      scope.model[deepLinkField.key] = 'brown'
      scope.$digest()
      expect(compileAndDigest).to.not.throw()
      expect(deepLinkField.watcher.listener).to.have.been.called
    })

Comment (kentcdodds) [inline src/directives/formly-form.test.js]: Not quite. We don't need to compile and digest more than once. But the other stuff is probably good.

**Comment by kentcdodds** [comment]
Also, the commit message isn't quite right. Please review [the commit message convention](https://github.com/ajoslin/conventional-changelog/blob/master/conventions/angular.md). Thanks again!

**Comment by MCKRUZ** [comment]
I didn’t commit anything though…I was waiting on your assessment of the test stuff I sent you.

**Pull request closed without merging**